exporter/otlpexporter/config.go

```diff
@@ -34,7 +33,29 @@ type Config struct {
 
 	configgrpc.GRPCClientSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
 
-	Arrow *arrow.Settings `mapstructure:"arrow"`
+	IncludeClientMetadataSettings `mapstructure:"include_client_metadata"`
+
+	Arrow ArrowSettings `mapstructure:"arrow"`
+}
+
+// IncludeClientMetadataSettings lists client.Metadata keys to copy to outgoing
+// request contexts.
+//
+// Note: this probably belongs in a common location because it probably will
```

Comment:
I think we should just accept everything that is in the outgoing grpc context. That would be make the arrow stream compatible with all collector's extensions that uses the [auth](https://github.com/open-telemetry/opentelemetry-collector/tree/main/extension/auth) extension, including the `header_setter` extension.

Reply:
Oh I see. I have missed the `header_setter` extension. This code base is not well documented omg.

Reply:
Updated to use the Auth extension mechanism so that headersetter_extension will work.